extensions/twenty/src/create-people.tsx

```diff
@@ -0,0 +1,138 @@
+import { Action, ActionPanel, Form, List, showToast, Toast, useNavigation } from "@raycast/api";
+import { FormValidation, useForm } from "@raycast/utils";
+import axios from "axios";
+import { useState } from "react";
+import { useAuthHeaders } from "./hooks/use-auth-headers";
+import { useGetCompanies } from "./hooks/use-company";
+import ListPeople from "./list-people";
+
+interface CreatePersonFormProps {
+  firstName: string;
+  lastName: string;
+  primaryEmail: string;
+  primaryPhoneNumber?: string;
+  linkedinLink?: string;
+  jobTitle?: string;
+  city?: string;
+  companyId?: string;
+}
+
+export default function CreatePersonForm() {
+  const { push } = useNavigation();
+  const { companies, isLoading, error, hasMore, loadMore } = useGetCompanies();
+  const [creationIsLoading, setCreationIsLoading] = useState(false);
+
+  if (error) {
+    return <List.EmptyView title="Error" description={error.message} />;
+  }
+
+  const { handleSubmit, itemProps, setValue } = useForm<CreatePersonFormProps>({
+    async onSubmit(values) {
+      if (!isLoading || !creationIsLoading) {
+        setCreationIsLoading(true);
+        await createPerson(values);
+        setCreationIsLoading(false);
+        push(<ListPeople />);
```

Comment:
If the `createPerson` function fails, we get a Failure Toast but the `ListPeople` View is still pushed onto the Stack which seems counter intuitive. I suggest that if the function fails, you can push a separate ErrorView and display the errors in a Detail View and/or via Toast.

Here's a response I got:
`{
  "statusCode": 400,
  "messages": [
    "Invalid value '' at 'data.companyId'; Expected type 'UUID'. Invalid UUID"
  ],
  "error": "Bad Request"
}`
You could cycle through the Messages and inform user what is wrong.


Another issue is if I choose CompanyID as "Empty", the function fails 😔.